sycl/source/detail/scheduler/graph_processor.cpp

```diff
@@ -71,6 +71,12 @@ bool Scheduler::GraphProcessor::enqueueCommand(Command *Cmd,
       return false;
   }
 
+  for (const EventImplPtr &Event : Cmd->getPreparedHostDepsEvents()) {
```

Comment:
Can you please explain the rationale behind this change? Maybe a comment? 
Also, I think it would be great to have a small unit-test in `sycl/unittests/scheduler/BlockedCommands.cpp` that exposes the problem and the fix.

Reply:
I've added a comment.